### PR TITLE
WIP Added another SDK resolver that will find the snap installation

### PR DESCRIPTION
**Pull request by wheelerlaw**
Create a new resolver capable of finding installations by Snap. 

Just need to modify he build.cake scipts to copy it into the correct location so that its picked up by the SdkResolverLoader. 

Fixes OmniSharp/omnisharp-vscode#3160

### CODE REVIEW

**Comment by nguerrera** [comment]
See https://github.com/OmniSharp/omnisharp-vscode/issues/3160#issuecomment-517325131

File: src/OmniSharp.MSBuild/SnapSdkResolver.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+namespace Microsoft.Build.BackEnd.SdkResolution
+{
+    /// <summary>
+    ///     Resolve SDKs that are part of a Snap installation
+    /// <remarks>
+    ///     This will search for the Snap SDK installation of the .NET Core SDK
+    ///     by executing `dotnet --list-sdks`.
+    /// </remarks>
+    /// </summary>
+    internal class SnapSdkResolver : SdkResolverBase
```
Comment (david-driscoll) [inline src/OmniSharp.MSBuild/SnapSdkResolver.cs]: Is this picked up just passively by MSBuild (by inheriting from `Microsoft.Build.Framework.SdkResolver`), or does it require additional registration in MEF?

**Comment by filipw** [comment]
I think the ultimate solution would be to fix it in the CLI https://github.com/dotnet/cli/issues/12110

**Comment by filipw** [comment]
Thanks for the PR. We will close this as it needs to be solved externally in the dotnet CLI.
The instructions for workarounds have been published here https://github.com/OmniSharp/omnisharp-vscode/wiki/Troubleshooting:-'The-.NET-Core-SDK-cannot-be-located.'-errors#linux-snap-instructions

**Pull request closed without merging**